src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,95 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_CAPACITY = 10;
+    private T[] list;
+    private int size;
+
+    public ArrayList() {
+        list = (T[]) new Object[DEFAULT_CAPACITY];
+    }
+
     @Override
     public void add(T value) {
-
+        if (size == list.length) {
+            grow();
+        }
+        list[size++] = value;
     }
 
     @Override
     public void add(T value, int index) {
-
+        if (index == size) {
+            add(value);
+            return;
+        }
+        checkIndex(index);
+        add(list[size - 1]);
```

Comment:
```suggestion
```
why are you adding that element separately, you can move with System.Arraycopy on 30s line

Reply:
if array is full there must be an error
description in System.Arraycopy:
"Otherwise, if any of the following is true, an IndexOutOfBoundsException is thrown and the destination is not modified:
...
**_destPos+length is greater than dest.length, the length of the destination array._**"

Reply:
so I adding element first

Reply:
![image](https://github.com/mate-academy/jv-array-list/assets/85360060/48458126-174c-4dc8-9c1b-ec03d4757f6f)

Reply:
i can add "value" and rewrite it with System.Arraycopy
```suggestion
        add(value);
        System.arraycopy(list, index, list, index + 1, size - index - 1);
        list[index] = value;
```
is this what you want?

Reply:
![image](https://github.com/mate-academy/jv-array-list/assets/85360060/cd8bbb20-bbc7-44cb-8d9f-47f04169876e)

Reply:
you forget to grow() if needed